feat/tests/test_fast_detector.py

```diff
@@ -35,3 +44,225 @@ def test_detect_image(self, single_face_img):
+    ):
+        """
+        Make sure that when the same images are passed in with and without batch
+        processing, the detected landmarks and poses come out to be the same
+        """
+        # Each sublist of images contains different sizes
+        all_images = (
+            [single_face_img] + [multi_face_img] + multiple_images_for_batch_testing
+        )
+
+        # Multiple images with different sizes are ok as long as batch_size == 1
+        # Detections will be done in each image's native resolution
+        det_output_default = self.detector.detect(inputs=all_images, batch_size=1)
+
+        # TODO: fast_detector does not currently raise ValueError
+        # If batch_size > 1 then output_size must be set otherwise we can't stack to make a
+        # tensor
+        # with pytest.raises(ValueError):
+        #     _ = self.detector.detect(
+        #         inputs=all_images,
+        #         batch_size=5,
+        #     )
+
+        # Here we batch by resizing each image to 256xpadding
+        batched = self.detector.detect(
+            inputs=all_images, batch_size=5, output_size=256
+        )
+
+        # We can also forcibly resize images even if we don't batch process them
+        nonbatched = self.detector.detect(
+            inputs=all_images, batch_size=1, output_size=256
+        )
+
+        # To make sure that resizing doesn't interact unexpectedly with batching, we should
+        # check that the detections we get back for the same sized images are the same when
+        # processed as a batch or serially. We check each column separately
+        au_diffs = np.abs(batched.aus - nonbatched.aus).max()
+        TOL = 0.5
+        bad_aus = au_diffs[au_diffs > TOL]
+        if len(bad_aus):
+            warnings.warn(
+                f"Max AU deviation is larger than tolerance ({TOL}) when comparing batched vs non-batched detections: {bad_aus}"
+            )
+        else:
+            print(
+                f"Max AU deviation (batched - nonbatched): {au_diffs.idxmax()}: {au_diffs.max()}"
+            )
+            
+    def test_fast_empty_init():
+        """Should fail if not models provided"""
+        with pytest.raises(ValueError):
+            _ = FastDetector(
+                emotion_model=None,
+                au_model=None,
+                landmark_model=None,
+                identity_model=None,
+            )
+    
+    def test_fast_init_with_wrongmodelname():
+        """Should fail with unsupported model name"""
+        with pytest.raises(ValueError):
+            _ = FastDetector(emotion_model="badmodelname")
+    
+    def test_fast_nofile(self):
+        """Should fail with missing data"""
+        with pytest.raises((FileNotFoundError, RuntimeError)):
+            inputFname = os.path.join(get_test_data_path(), "nosuchfile.jpg")
+            _ = self.detector.detect(inputFname)
+    
+    
+    # no face images
+    #TODO: figure out why all failing
```

Comment:
It looks like we aren't handling a lack of faces during `FastDetector.forward()`. The culprit is that on line 678 of `FastDetector.py` we need to be checking if `n_faces == 0`. If so, then we need to run the `else` blocks of all the detectors below. This will require a little reshuffling, but essentially if there are no faces we should just build `nan` tensors for each detector of the correct dimensions, e.g like we're doing on line 692 for the landmarks, 704 for the emotions, etc. If you want to take a shot at this feel free, or I can implement the changes as well.